client/src/components/DetailView.js

```diff
@@ -0,0 +1,383 @@
+  margin: 0px 5px;
+  cursor: default;
+`;
+
+const AllActorsContainer = styled.div`
+  display: flex;
+  flex-flow: row nowrap;
+  align-items: baseline;
+  justify-content: center;
+  margin: 10px 10px;
+  flex: 1 1;
+  width: auto;
+`;
+
+const ActorContainer = styled.div`
+  display: flex;
+  flex-flow: column nowrap;
+  justify-content: flex-start;
+  align-items: center;
+  text-align: center;
+  padding-top: 10px;
+  font: 300 0.7rem 'Roboto', sans-serif;
+  color: #aeb2f5;
+  flex: 0 0 25%;
+  height: auto;
+  word-break: break-word;
+  margin: 0px 2.5px;
+`;
+
+const ActorImageWrapper = styled.div`
+  border: hidden;
+  border-radius: 10px;
+  box-shadow: 0px 0px 6px #1e194f;
+  overflow: hidden;
+  min-width: 30px;
+  min-height: 80px;
+  flex: 1 1;
+  margin: 5px;
+
+  & > img {
+    height: 100%;
+    width: 100%;
+    object-fit: cover;
+    object-position: 50% 50%;
+    border: hidden;
+    border-radius: 15px;
+    margin-bottom: -3px;
+  }
+`;
+
+const WatchlistCheckContainer = styled.div`
+  display: flex;
+  flex-flow: row nowrap;
+  justify-content: space-evenly;
+  padding-bottom: 10px;
+  margin: 5px 10px;
+`;
+
+const WatchListCheckLabel = styled.label`
+  display: flex;
+  flex-flow: column nowrap;
+  align-items: center;
+  margin: 10px;
+  cursor: pointer;
+
+  & > * {
+    margin: 2.5px 0px;
+  }
+`;
+
+const WatchListCheck = styled.input`
+  position: absolute;
+  opacity: 0;
+  cursor: pointer;
+  height: 0;
+  width: 0;
+  user-select: none;
+`;
+
+const WatchListCheckText = styled.div`
+  display: block;
+  font-style: ${(props) => (props.checked ? 'italic' : 'normal')};
+  font-size: 0.7rem;
+  font-weight: 200;
+  font-family: 'Roboto', sans-serif;
+  color: #e7eaff;
+  background: transparent;
+`;
+
+function DetailView({
+  showPoster,
+  showTitle,
+  showRating,
+  showOverview,
+  showGenres,
+  showActors,
+}) {
+  const [watchlistAction, setWatchlistAction] = useState(null);
+
+  const addedToWatchlist = watchlistAction === 'addToWatchlist';
+  const addedToWatched = watchlistAction === 'addToWatched';
+
+  function handleWatchlistActionClick(event) {
+    const targetWatchlistAction = event.target.value;
+
+    if (watchlistAction === targetWatchlistAction) {
+      setWatchlistAction(null);
+    } else {
+      setWatchlistAction(targetWatchlistAction);
+    }
+  }
+  return (
+    <Container>
+      <CloseButton>
+        <img src={CloseIcon} alt="small icon with an x symbol" />
+      </CloseButton>
+      <PosterContainer>
+        <img src={showPoster} alt="poster of the tv show" />
+      </PosterContainer>
+      <TitleContainer>
+        <Rating>
+          <img src={RatingIcon} alt="star icon to represent rating" />
+          {`${showRating} %`}
+        </Rating>
+        <ShowTitle>{showTitle}</ShowTitle>
+        <TrailerButton>
+          <img src={TrailerIcon} alt="a play icon to represent a video" />
+          Watch the Trailer
+        </TrailerButton>
+      </TitleContainer>
```

Comment:
This could also be a separate component that receives props like `rating` and `title`